src/ai/backend/manager/registry.py

```diff
@@ -405,29 +405,31 @@ async def create_session(
         current_task = asyncio.current_task()
         assert current_task is not None
 
-        # Check work directory and reserved name directory.
-        mount_map = config.get("mount_map")
+        mount_id_map = config.get("mount_id_map", {})
+        mount_map = config.get("mount_map", {})
 
-        if mount_map is not None:
-            original_folders = mount_map.keys()
-            alias_folders = mount_map.values()
-            if len(alias_folders) != len(set(alias_folders)):
-                raise InvalidAPIParameters("Duplicate alias folder name exists.")
+        combined_mount_map = {**mount_map, **mount_id_map}
 
-            alias_name: str
-            for alias_name in alias_folders:
-                if alias_name is None:
-                    continue
-                if alias_name.startswith("/home/work/"):
-                    alias_name = alias_name.replace("/home/work/", "")
-                if alias_name == "":
-                    raise InvalidAPIParameters("Alias name cannot be empty.")
-                if not verify_vfolder_name(alias_name):
-                    raise InvalidAPIParameters(str(alias_name) + " is reserved for internal path.")
-                if alias_name in original_folders:
-                    raise InvalidAPIParameters(
-                        "Alias name cannot be set to an existing folder name: " + str(alias_name)
-                    )
+        # Check work directory and reserved name directory.
+        original_folders = combined_mount_map.keys()
+        alias_folders = combined_mount_map.values()
+        if len(alias_folders) != len(set(alias_folders)):
+            raise InvalidAPIParameters("Duplicate alias folder name exists.")
```

Comment:
A situation where UUID and name point to the same vfolder is expected, but we will correct this next time.